Decks/Source/LimitedDeck.cpp

```diff
@@ -0,0 +1,47 @@
+#include "LimitedDeck.hpp"
+
+HandDeck::HandDeck(int maxSize)
+:_maxSize(maxSize)
+{
+
+}
+
+TableDeck::TableDeck(int maxSize)
+:_maxSize(maxSize)
+{
+
+}
+
+void HandDeck::drawCard(Deck &deck)
+{
+  if(_size >= _maxSize)
+  {
+    std::cout << "Vaše ruka je plná" << std::endl;
+  }
+  else
+  {
+    _cards.push_back(deck._cards.at(0));
+    deck._cards.erase(deck._cards.begin());

```

Comment:
Funkcja w Decku (i każdym kolejnym ewentualnie przeciążona), która usuwa kartę vectora i zwraca do niej unique_ptr i wtedy w tej funkcji to wywołujemy. Wtedy też _cards wchodzi do private, a nie public. Hermetyzacja danych!